src/App.tsx

```diff
@@ -1,223 +1,181 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Condition } from './types/Condition';
+import { Todo } from './types/Todo';
+import { addTodos, deleteTodos, getTodos } from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isError, setIsError] = useState('');
+  const [filterType, setFilterType] = useState(Condition.All);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+  const [activeLoading, setActiveLoading] = useState<number[]>([]);
+  const [loader, setLoader] = useState<number>(0);
+
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const filteredTodos = todos.filter(todo => {
+    switch (filterType) {
+      case Condition.Active:
+        return !todo.completed;
+      case Condition.Completed:
+        return todo.completed;
+      default:
+        return Condition.All;
+    }
+  });
+
+  const loadApiTodos = () => {
+    if (user) {
+      getTodos(user.id)
+        .then(res => (
+          (res)
+            ? setTodos(res)
+            : setIsError('Unable to add a todo')
```

Comment:
Use the catch method to catch the error. BTW would be great to rewrite async functions with async/await. But it's up to you